\documentclass[a4paper]{article}
\usepackage[OT1]{fontenc}
\usepackage{Sweave}
\begin{document}

\title{FunciSNP: Functional Identification of SNPs with\\Phenotype by 
Coincidence with Chromatin Biofeatures\\Vignette}
\author{Simon G. Coetzee and Houtan Noushmehr\\\\Norris Cancer Center\\Keck 
School of Medicine\\University of Southern California\\Los Angeles, USA.\\\\
and\\\\Faculdade de Medicina de Ribeir\^{a}o Preto\\Departmento de 
Gen\'{e}tica\\Universidade de S\^{a}o Paulo\\Ribeir\^{a}o Preto, S\^{a}o Paulo, 
Brasil}

\maketitle
\section*{Introduction}
%include intro text

\section*{Load FunciSNP+other useful libraries}
<<packages>>=
#When package is offically posted in Bioconductor, uncomment next 2 lines.
#source("http://bioconductor.org/biocLite.R")
#biocLite("FunciSNP");
## Following two packages and options() are not required to run 'FunciSNP' but 
#will enhance the analysis experience.
#library(setwidth); ## Automatically set the value of options("width") when the 
#terminal emulator is resized
#library(colorout); ## colorize R output on terminal emulators
options(width=80);
##FunciSNP library and other related libraries needed.
library("org.Hs.eg.db");
library("gplots");
library("gtools");
library("ggplot2");
library("matlab");

library(FunciSNP);
package.version("FunciSNP");
@

\section*{Identify Func-y-SNP using published GWAS SNPs and publicly available 
biological features (ENCODE ChIPseq peaks)}
\subsection*{FunciSNP()}
This is the main funtion of FunciSNP. It will identify correlated SNPs which are
 in linkage disequilibrium (LD) to a known disease associated tagSNP. It will 
also determine if the correlated SNP in LD to the tagSNP overlaps a genomic 
biological feature. Correlated SNPs are directly imported from the current 
public release of the 1000 genomes database. 1000 genomes ftp servers available 
for the 1000 genomes public data: 1) National Center for Biotechnology 
Information (NCBI) ftp://ftp-trace.ncbi.nih.gov/1000genomes/; 2) European 
Bioinformatics Institute (EBI) ftp://ftp.1000genomes.ebi.ac.uk/vol1/.

Correlated SNPs in LD to a tagSNP and overlapping genomic biological features 
are known as putative functional SNPs (also defined as 'Func-y-SNP' elsewhere in
 the package.).

As an example, we collected SNPs identified by GWAS for Glioblastoma multiforme 
(GBM). In this example, GBM includes lower grade glioma, thus we label all 
objects with 'glioma'.
<<>>=
## Full path to the example GWAS SNP regions file for Glioblastoma 
#  (collected from SNPedia on Jan 2012)
glioma.snp <- file.path(system.file('data', package='FunciSNP'), 
dir(system.file('data',package='FunciSNP'), pattern='.snp$'));
glioma.snp;
## Full path to the example biological features BED files 
#  derived from the ENCODE project for Glioblastoma U-87 cell lines.
glioma.bio <- system.file('extdata',package='FunciSNP');
glioma.bio;
## FunciSNP analysis, extracts correlated SNPs from the 
#  1000 genomes db ("ncbi" or "ebi") and finds overlaps between 
#  correlated SNP and biological features and then 
#  calculates LD (Rsquare, Dprime, distance, p-value).
## Will take more than 5 min to run. Depending on number of CPUs and internet connection.
#glioma <- FunciSNP(snp.regions.file=glioma.snp, bio.features.loc = glioma.bio, 
#bio.features.TSS=FALSE);
#glioma;
#summary(glioma);

## The glioma example set can also be called by the following line if you don't 
#want to redo the initial analysis. 
data(glioma);
glioma;
summary(glioma);
@

\subsection*{FunciSNPAnnotateSummary()}
All known genomic features (exon, intron, 5'UTR, 3'UTR, promoter, lincRNA or in 
gene desert (intergentic)) are used to annotate the newly identified Func-y-SNP.
 Information described in this data.frame is used for all summary plots, table, 
 and bed file generations.
<<>>=
glioma.anno <- FunciSNPAnnotateSummary(glioma);
summary(glioma.anno);
dim(glioma.anno)
head(glioma.anno)
names(glioma.anno)
@
\subsection*{FunciSNPtable()}
Using a specified Rsquare value (0-1) to subset the data, a table is generated 
which summarizes the total number of Func-y-SNPs, associated tagSNPs, and number
 of overlapping biofeatures. This will provide user a first look at the total 
 number of available Func-y-SNP at a particular Rsquare cutoff.  If geneSum is 
 set to TRUE, a list of gene names is reported instead.
<<>>=
FunciSNPtable(glioma.anno, rsq=0.5);
FunciSNPtable(glioma.anno, rsq=0.5, geneSum=T);
@
\subsection*{FunciSNPsummaryOverlaps()}
<<>>=
FunciSNPsummaryOverlaps(glioma.anno)
FunciSNPsummaryOverlaps(glioma.anno, rsq=0.5)
@
\subsection*{FunciSNPidsFromSummary()}
<<>>=
FunciSNPidsFromSummary(glioma.anno, tagsnpid="rs6010620", num.features=2, 
rsq=0.5)
@

\subsection*{FunciSNPplot()}
FunciSNPplot is a function developed to plot various types of plots to summarize
 and assist end-user in making informed discoveries of FunciSNP results. Plots 
 can be stored in a folder for future reference.
<<>>=
pdf("glioma_dist.pdf")
FunciSNPplot(glioma.anno)
dev.off()
@
\begin{figure}[ht!]
\begin{center}
\includegraphics{glioma_dist.pdf}
\caption{\label{fig:glioma_dist.pdf} Distribution of Rsquare values of all 
Correlated SNPs}
{\footnotesize{}}
\end{center}
\end{figure}

<<>>=
FunciSNPplot(glioma.anno, splitbysnp=T)
ggsave("glioma_dist_bysnp.pdf")
@
\begin{figure}[ht!]
\begin{center}
\includegraphics{glioma_dist_bysnp.pdf}
\caption{\label{fig:glioma_dist_bysnp.pdf} Distribution of Rsquare values of all
 Correlated SNPs divided by the tagSNP and it's location.}
{\footnotesize{}}
\end{center}
\end{figure}

<<>>=
pdf("glioma_genomic_sum_rcut.pdf")
FunciSNPplot(glioma.anno, rsq=0.5, genomicSum=T, save=F)
dev.off()
@
\begin{figure}[ht!]
\begin{center}
\includegraphics{glioma_genomic_sum_rcut.pdf}
\caption{\label{fig:glioma_genomic_sum_rcut.pdf} Stacked bar chart summarizing 
all correlated SNPs for each of the identified genomie features: exon, intron, 
5UTR, 3UTR, promoter, lincRNA or in gene desert. Rsquare cutoff
 at 0.5. This plot is most informative if used with a rsq value.}
{\footnotesize{}}
\end{center}
\end{figure}

Will output two plots per biofeature. The first plot is a scatter plot showing 
the relationship between Rsquare and Distance to tagSNP for each Func-y-SNP. The
 second plot is a histogram distribution of number of correlated SNPs at each 
 Rsquare value. Each set of plot is further divided by tagSNP. Best if used with
  rsq value.
<<>>=
## Following will output a series of plots for each biofeature at rsq=0.5
FunciSNPplot(glioma.anno, tagSummary=T, rsq=0.5)
@
\begin{figure}[ht!]
\begin{center}
\includegraphics{FunciSNP.0.1.7/plots/TFBS_Pol2_U87_R2vsDist_riskSNP.pdf}
\caption{\label{fig:TFBS_Pol2_U87_R2vsDist_riskSNP.pdf} Scatter plot 
showing the relationship between Rsquare and Distance to tagSNP for each 
Func-y-SNP}
{\footnotesize{}}
\end{center}
\end{figure}

\begin{figure}[ht!]
\begin{center}
\includegraphics{FunciSNP.0.1.7/plots/TFBS_Pol2_U87_R2summary_riskSNP.pdf}
\caption{\label{fig:TFBS_Pol2_U87_R2summary_riskSNP.pdf} Histogram 
distribution of number of correlated SNPs at each Rsquare value}
{\footnotesize{}}
\end{center}
\end{figure}


\subsection*{FunciSNPbed()}
FunciSNPbed outputs a unique BED file which can be used to view in any genomic 
browser compatible with BED formats. To learn more about BED formats, see UCSC 
Genome Browser FAQ (http://genome.ucsc.edu/FAQ/FAQformat). Each tagSNP 
which is in LD to a corresponding Func-y-SNP overlapping at least one biofeature
 is colored black, while the Func-y-SNP is colored red. The initial position is 
 provided by the first tagSNP and the first linked Func-y-SNP. We recommend 
 using UCSC genome browser to view your BED files. This is useful so you can 
 view all public and private tracks in relation to FunciSNP results.
<<>>=
## will output to current working directory.
FunciSNPbed(glioma.anno, rsq=0.5);
@

\newpage

<<sessionInfo>>=
sessionInfo()
@

\end{document}
