% 
\documentclass[a4paper]{article}
\usepackage[OT1]{fontenc}
\usepackage{Sweave}
\begin{document}

\title{TCGA COAD/READ DNA Methylation\\ Data Analysis}
\author{Toshinori Hinoue, Daniel J. Weisenberger and Peter W. Laird\\ \\ USC Epigenome Center}

\maketitle
\section*{Introduction}
This document describes DNA methylation data analysis performed in the TCGA manuscript on Colorectal cancer entitled "Comprehensive Molecular Characterization of Human Colon and Rectal tumors". We used the Illumina Infinium DNA methylation platform (HumanMethylation27 BeadChip), which interrogates 27,578 CpG sites located in proximity to the transcription start sites of 14,475 genes.


\section*{Required libraries and function}
<<packages>>=
library(RPMM)
library(seriation)
library(RColorBrewer)
library(ggplot2)
source("./heatmap.custom.R")
@


\section*{TCGA COAD level 3 DNA methylation data}
<<echo= false, results =hide>>=
#TempPath <- "/home/toshi/Documents/vm/TCGA/01 ColonRectum/03 Data/TCGA archive data/DNA methylation/jhu-usc.edu_COAD.HumanMethylation27.Level_3"  # linux
TempPath <- "/Users/Toshi/Documents/TCGA/01_COADREAD/03 Data/TCGA archive data/DNA methylation/jhu-usc.edu_COAD.HumanMethylation27.Level_3" # mac
options(width=70)
@

All COAD (Colon Adenocarcinoma) DNA methylation data were downloaded from the TCGA Data Portal. The following data archives were used for the analyses in the manuscript:
<<directries>>=
dirs=list.files(TempPath)
dirs
@

\subsection*{Extract COAD DNA methylation $\beta$-values from each data archive}
<<COAD All>>=
betaAll <-  vector("list",length(grep('Level_3',dirs)))
names(betaAll) <- dirs[grep('Level_3',dirs)]
ngenes <- 27578

for (i in seq(betaAll)) {
	SamplePath <- file.path(TempPath,dirs[i])
	filelist <- list.files(SamplePath, pattern = "lvl-3")
	beta <- matrix(0,ngenes,length(filelist))
	tcga.id <- vector("character",length(filelist))
	data1 <- read.delim(file=file.path(SamplePath,filelist[1]),sep="\t", row.names=1,skip=1,header=T)[1]
	row.names(beta) <- row.names(data1)
	beta[,1] <- data1[,1]
	tcga.id[1] <- strsplit(readLines(file.path(SamplePath,filelist[1]),n=1),"\t")[[1]][2]
	for (j in 2:length(filelist)) {
		beta[,j] <- read.delim(file.path(SamplePath,filelist[j]),sep="\t",row.names=1,skip=1,header=T)[1]$Beta_Value
		tcga.id[j]<- strsplit(readLines(file.path(SamplePath,filelist[j]),n=1),"\t")[[1]][2]}
	dimnames(beta)[[2]] <- tcga.id
	betaAll[[i]] <- beta
}
summary(betaAll)
@

\subsubsection*{Level$\_$3.1.2.0 - Batch 28}
31 tumors and 17 adjacent-normal samples 
<<COAD Batch 28>>=
dim(betaAll[[1]])
dimnames(betaAll[[1]])[[2]]
@

\subsubsection*{Level$\_$3.2.2.0 - Batch 29} 
7 tumors and 5 adjacent-normal samples
<<COAD Batch 29>>=
dim(betaAll[[2]])
dimnames(betaAll[[2]])[[2]]
@

\subsubsection*{Level$\_$3.3.1.0 - Batch 30}
15 tumors 
<<COAD Batch 30>>=
dim(betaAll[[3]])
dimnames(betaAll[[3]])[[2]]
@

\subsubsection*{Level$\_$3.4.1.0 - Batch 33} 
17 tumors
<<COAD Batch 33>>=
dim(betaAll[[4]])
dimnames(betaAll[[4]])[[2]]
@

\subsubsection*{Level$\_$3.5.2.0 - Batch 36} 
31 tumors
<<COAD Batch 36>>=
dim(betaAll[[5]])
dimnames(betaAll[[5]])[[2]]
@

\subsubsection*{Level$\_$3.6.0.0 - Batch 41} 
48 tumors and 4 adjacent-normal samples
<<COAD Batch 41>>=
dim(betaAll[[6]])
dimnames(betaAll[[6]])[[2]]
@

\noindent We removed TCGA-AA-3970-01 since it was run in duplicate in Batch 45. 
<<COAD Batch 41.2>>=
betaAll[[6]] <- betaAll[[6]][,-41]
dim(betaAll[[6]])
@

\subsubsection*{Level$\_$3.7.0.0 - Batch 45} 
6 tumors 
<<COAD Batch 45>>=
dim(betaAll[[7]])
dimnames(betaAll[[7]])[[2]]
@

\subsubsection*{Level$\_$3.8.0.0 - Batch 66} 
13 tumors and 11 adjacent-normal samples
<<COAD Batch 66>>=
dim(betaAll[[8]])
dimnames(betaAll[[8]])[[2]]
@

\subsection*{Combine COAD DNA methylation $\beta$-values from all the data archives}
<<COAD data merge>>=
COAD_beta <- betaAll[[1]]
for (i in 2:(length(betaAll))) {
	COAD_beta <- cbind(COAD_beta, betaAll[[i]])
}	
@

<<COAD data T and N>>=
dim(COAD_beta) 
# The number of COAD tumor samples
length(grep("^.{13}01",dimnames(COAD_beta)[[2]]))
# The number of COAD adjacent-normal samples
length(grep("^.{13}11",dimnames(COAD_beta)[[2]]))
@

\subsection*{Create a COAD sample annotation file}
<<COAD sample annotation merge>>=
sampleID <- NULL
for (i in seq(betaAll)) sampleID <-  c(sampleID, dimnames(betaAll[[i]])[[2]])
Batch <- c(rep("28",dim(betaAll[[1]])[[2]]),rep("29",dim(betaAll[[2]])[[2]]),
		rep("30",dim(betaAll[[3]])[[2]]),rep("33",dim(betaAll[[4]])[[2]]),
		rep("36",dim(betaAll[[5]])[[2]]),rep("41",dim(betaAll[[6]])[[2]]),
		rep("45",dim(betaAll[[7]])[[2]]), rep("66",dim(betaAll[[8]])[[2]]))
Location <- rep("Colon", length(sampleID))
TissueType <- rep("Tumor", length(sampleID))
TissueType[grep("^.{13}20",dimnames(COAD_beta)[[2]])] <- "Control"
TissueType[grep("^.{13}11",dimnames(COAD_beta)[[2]])] <- "Normal"
# Tissue source sites
TSS <-  substring(sampleID,6,7) 
COAD_pdata <- data.frame(sampleID,Batch,Location,TissueType,TSS)
head(COAD_pdata)
@

\section*{TCGA READ level 3 DNA methylation data}
<<echo= false, results =hide>>=
#TempPath <- "/home/toshi/Documents/vm/TCGA/01 ColonRectum/03 Data/TCGA archive data/DNA methylation/jhu-usc.edu_READ.HumanMethylation27.Level_3" # linux
TempPath <- "/Users/Toshi/Documents/TCGA/01_COADREAD/03 Data/TCGA archive data/DNA methylation/jhu-usc.edu_READ.HumanMethylation27.Level_3" # mac
@

\noindent All READ (Rectal Adenocarcinoma) DNA methylation data were downloaded from the TCGA Data Portal. The following data archives were used for the analyses in the manuscript:
<<directries>>=
dirs=list.files(TempPath)
dirs
@

\subsection*{Extract READ DNA methylation $\beta$-values from each data archive}
<<READ All>>=
betaAll <-  vector("list",length(grep('Level_3',dirs)))
names(betaAll) <- dirs[grep('Level_3',dirs)]
ngenes <- 27578

for (i in seq(betaAll)) {
	SamplePath <- file.path(TempPath,dirs[i])
	filelist <- list.files(SamplePath, pattern = "lvl-3")
	beta <- matrix(0,ngenes,length(filelist))
	tcga.id <- vector("character",length(filelist))
	data1 <- read.delim(file=file.path(SamplePath,filelist[1]),sep="\t", row.names=1,skip=1,header=T)[1]
	row.names(beta) <- row.names(data1)
	beta[,1] <- data1[,1]
	tcga.id[1] <- strsplit(readLines(file.path(SamplePath,filelist[1]),n=1),"\t")[[1]][2]
	for (j in 2:length(filelist)) {
		beta[,j] <- read.delim(file.path(SamplePath,filelist[j]),sep="\t",row.names=1,skip=1,header=T)[1]$Beta_Value
		tcga.id[j]<- strsplit(readLines(file.path(SamplePath,filelist[j]),n=1),"\t")[[1]][2]}
	dimnames(beta)[[2]] <- tcga.id
	betaAll[[i]] <- beta
}
summary(betaAll)
@

\subsubsection*{Level$\_$3.1.2.0 - Batch 42 set 1} 
16 tumors and 4 adjacent-normal samples
<<READ Batch 42.1>>=
dim(betaAll[[1]])
dimnames(betaAll[[1]])[[2]]
@

\subsubsection*{Level$\_$3.2.1.0 - Batch 42 set 2} 
10 tumors
<<READ Batch 42.2>>=
dim(betaAll[[2]])
dimnames(betaAll[[2]])[[2]]
@

\subsubsection*{Level$\_$3.3.1.0 - Batch 42 set 3} 
9 tumors 
<<READ Batch 42.3>>=
dim(betaAll[[3]])
dimnames(betaAll[[3]])[[2]]
@

\subsubsection*{Level$\_$3.4.1.0 - Batch 42 set 4} 
9 tumors 
<<READ Batch 42.4>>=
dim(betaAll[[4]])
dimnames(betaAll[[4]])[[2]]
@

\subsubsection*{Level$\_$3.5.1.0 - Batch 42 set 5} 
8 tumors 
<<READ Batch 42.5>>=
dim(betaAll[[5]])
dimnames(betaAll[[5]])[[2]]
@

\subsubsection*{Level$\_$3.6.0.0 - Batch 46} 
18 tumors and 1 adjacent-normal sample
<<READ Batch 46>>=
dim(betaAll[[6]])
dimnames(betaAll[[6]])[[2]]
@

\noindent We removed TCGA-AG-3728-01 since it was run in duplicate in Batch 36. 
<<READ Batch 46.2>>=
betaAll[[6]] <- betaAll[[6]][,-4]
dim(betaAll[[6]])
@

\subsection*{Combine READ DNA methylation $\beta$-values from all the data archives}
<<READ data merge>>=
READ_beta <- betaAll[[1]]
for (i in 2:(length(betaAll))) {
	READ_beta <- cbind(READ_beta, betaAll[[i]])
}
@

<<READ data T and N>>=
dim(READ_beta) 
# The number of READ tumor samples
length(grep("^.{13}01",dimnames(READ_beta)[[2]]))
# The number of READ adjacent-normal samples
length(grep("^.{13}11",dimnames(READ_beta)[[2]]))
@

\subsection*{Create a READ sample annotation file}
<<READ sample annotation merge>>=
sampleID <- NULL
for (i in seq(betaAll)) sampleID <-  c(sampleID, dimnames(betaAll[[i]])[[2]])
Batch <- c(rep("42.1",dim(betaAll[[1]])[[2]]),rep("42.2",dim(betaAll[[2]])[[2]]),
		   rep("42.3",dim(betaAll[[3]])[[2]]),rep("42.4",dim(betaAll[[4]])[[2]]),
		   rep("42.5",dim(betaAll[[5]])[[2]]),rep("46",dim(betaAll[[6]])[[2]]))
Location <- rep("Rectum", length(sampleID))
TissueType <- rep("Tumor", length(sampleID))
TissueType[grep("^.{13}20",dimnames(READ_beta)[[2]])] <- "Control"
TissueType[grep("^.{13}11",dimnames(READ_beta)[[2]])] <- "Normal"
# Tissue source sites
TSS <-  substring(sampleID,6,7) 
READ_pdata <- data.frame(sampleID,Batch,Location,TissueType,TSS)
head(READ_pdata)
@

\section*{Illumina HumanMethylation27 array description file (adf)}
<<adf>>=
MageTabPath <- file.path(TempPath,dirs[7])
list.files(MageTabPath)                               
file.adf <- list.files(MageTabPath, pattern = "adf")
adf.HM27 <- read.delim(file=file.path(MageTabPath, file.adf),sep="\t", row.names=1,header=T)
dim(adf.HM27)
names(adf.HM27)
@

\section*{Combine COAD and READ DNA methylation data}
\subsection*{DNA methylation $\beta$-values}
<<COAD READ data merge>>=
COADREAD_beta <- merge(COAD_beta,READ_beta,by="row.names",sort=FALSE)
row.names(COADREAD_beta) <- COADREAD_beta[[1]]
COADREAD_beta <- COADREAD_beta[,-1]
dim(COADREAD_beta)
@

\subsection*{A sample annotation file}
<<COAD READ pdata merge>>=
COADREAD_pdata <- rbind(COAD_pdata, READ_pdata)
COADREAD_pdata$sampleID <- as.character(COADREAD_pdata$sampleID)
dim(COADREAD_pdata)
@

\section*{Identification of DNA methylation-based subtypes using an unsupervised clustering analysis}
\noindent The unsupervised clustering analysis was performed only on the tumor samples:
<<RPMM-subset>>=
table(COADREAD_pdata$TissueType)
COADREAD_pdataT <- COADREAD_pdata[COADREAD_pdata$TissueType=="Tumor",]
COADREAD_betaT <- COADREAD_beta[,COADREAD_pdataT$sampleID]
@

\noindent Identified an NA-masked probe set (n=4,484). \\
We masked data points with NA from the probes that \\
1) contain known single nucleotide polymorphisms after comparison to the dbSNP database. \\
2) contain repetitive sequence elements that cover the targeted CpG locus in each 50 bp probe sequence. \\
3) are not uniquely aligned to the human genome at 20 nucleotides at the 3' terminus of the probe. sequence. \\
4) span known regions of small insertions and deletions in the human genome. \\

<<RPMM-masked probes>>=
count.na.row <- rowSums(is.na(COADREAD_betaT))
COADREAD_betaT$masked <- count.na.row==dim(COADREAD_betaT)[[2]]
table(COADREAD_betaT$masked)
@

\noindent Merged DNA methylation data and the adf.
<<RPMM-merge adf>>=
COADREAD_betaT.adf <- merge(adf.HM27, COADREAD_betaT, by="row.names")
row.names(COADREAD_betaT.adf) <- COADREAD_betaT.adf[[1]]
COADREAD_betaT.adf <- COADREAD_betaT.adf[,-1]
@

\noindent Removed probes which contain any NA-masked data points and probes that are designed for the sequences on X and Y chromosomes.  
<<RPMM-filter>>=
COADREAD_betaT.adf.fil <- subset(COADREAD_betaT.adf,masked=="FALSE" & 
								 Chr!="X" & Chr!="Y")
dim(COADREAD_betaT.adf.fil)
dat <- COADREAD_betaT.adf.fil[,COADREAD_pdataT$sampleID]
dat <- na.omit(dat)
dim(dat)
@

\noindent Identified probes that showed the top 10$\%$ (2,758) most variable DNA methylation levels based on the standard deviations across the colorectal tumor panel. 
<<RPMM-sd filtering>>=
dat$sdT<- apply(dat,1,sd,na.rm=TRUE)
sum(dat$sdT>=0.16864)
dat.2758 <- subset(dat, dat$sdT>=0.16864, select=c(1:236))
dim(dat.2758)
@

\noindent Performed a RPMM-based unsupervised clustering (see Supplemental Methods). \\
A fanny algorithm (a fuzzy clustering algorithm) was used for initialization and level-weighted version of Bayesian Information Criterion (BIC) as a split criterion for an existing cluster as implemented in the RPMM package. 
<<RPMM>>=
dat.2758 <- as.matrix(dat.2758)
rpmm <- blcTree(t(dat.2758), verbose=0, splitCriterion=blcSplitCriterionLevelWtdBIC)
rpmm
rpmmClass <- blcTreeLeafClasses(rpmm)
table(rpmmClass)

a <- lapply(levels(rpmmClass), function(k){dimnames(dat.2758)[[2]][which(rpmmClass==k)]})
names(a) <- levels(rpmmClass)
m <- matrix(as.character(unlist(a)))
clu <- matrix(c(rep(names(a)[[1]],length(a[[1]])),
				rep(names(a)[[2]],length(a[[2]])),
				rep(names(a)[[3]],length(a[[3]])),
				rep(names(a)[[4]],length(a[[4]]))))
asg <- data.frame(cbind(m,clu))
colnames(asg) <- c("sampleID","RPMM.Cluster")
@

\noindent Updated the sample annotation file with the RPMM-based DNA methylation clustering membership defined above.
<<RPMM-pdata>>=
COADREAD_pdata <- COADREAD_pdata[COADREAD_pdata$TissueType!="Control",]
dim(COADREAD_pdata)
COADREAD_pdata <- merge(COADREAD_pdata, asg, by="sampleID",all.x=TRUE)
head(COADREAD_pdata)
@


\section*{Heatmap visualization of DNA methylation data}
\noindent Here, we visualized the DNA methylation values that showed standard deviation >0.20 across 236 tumor samples. \\
We first performed seriation implemented in the \textit{seriation} package to determine the ordering of the tumor samples within each DNA methylation cluster. 
<<seriation>>=
sum(dat$sdT>=0.2)
dat.1403 <- subset(dat, sdT>=0.2,select=c(1:236))
dim(dat.1403)

pdata.temp <- subset(COADREAD_pdata,RPMM.Cluster!="NA")
dim(pdata.temp)
attach(pdata.temp, warn.conflicts = FALSE)
rLL <- as.character(pdata.temp[RPMM.Cluster=="rLL",1])
rLR <- as.character(pdata.temp[RPMM.Cluster=="rLR",1])
rRL <- as.character(pdata.temp[RPMM.Cluster=="rRL",1])
rRR <- as.character(pdata.temp[RPMM.Cluster=="rRR",1])
detach(pdata.temp)

clu1 <- as.matrix(dat.1403[,rLL])
clu2 <- as.matrix(dat.1403[,rLR])
clu3 <- as.matrix(dat.1403[,rRL])
clu4 <- as.matrix(dat.1403[,rRR])

set.seed(1022)
s1 <- seriate(clu1, margin=2)
s2 <- seriate(clu2, margin=2)
s3 <- seriate(clu3, margin=2)
s4 <- seriate(clu4, margin=2)

so1 <- get_order(s1)
so2 <- get_order(s2)
so3 <- get_order(s3)
so4 <- get_order(s4)
so <- c(names(so4),names(so3),names(so1),names(so2))

dat.1403.seriate <- dat.1403[,so]
@

\noindent Updated the sample annotation file with the sample order upon seriation.
<<seriation-pdata>>=
RPMM.Sample.order <- data.frame(names(dat.1403.seriate),seq(1:236))
names(RPMM.Sample.order) <- c("sampleID","seriate")
COADREAD_pdata <- merge(COADREAD_pdata, RPMM.Sample.order, by="sampleID", all.x=TRUE)
MC <- as.character(COADREAD_pdata$RPMM.Cluster) 
MC[is.na(MC)] <- "Normal"
cluster.names <- ifelse(MC=="rRR", "CIMP.H", 
		ifelse(MC=="rRL", "CIMP.L", 
				ifelse(MC=="rLL", "Cluster3",
						ifelse(MC=="rLR", "Cluster4", MC))))
table(cluster.names)
COADREAD_pdata$Meth.Cluster <- cluster.names
COADREAD_pdata <- COADREAD_pdata[,c(1,2,5,3,4,6,8,7)]
dim(COADREAD_pdata)
head(COADREAD_pdata)
@

\noindent Generated a heatmap.
<<heatmap>>=
dat <- as.matrix(dat.1403.seriate)
pdat <- COADREAD_pdata

vlookup <- function(val, df, col){df[df[1] == val,col]}
# DNA methylation cluster 
table(pdat$Meth.Cluster)
coltypeMC <- c("CIMP.H"="#a6cee3", "CIMP.L" = "#e3a5ce", "Cluster3"="#f4cc03","Cluster4"="#cee3a5")
ccMC <- apply(as.matrix(dimnames(dat)[[2]]), 1, vlookup, df=pdat, 
		col=which(names(pdat) == "Meth.Cluster"))
ccMC <- coltypeMC[as.character(ccMC)]

# Tumor location
table(pdat$Location)
coltypeST <- c("Colon"="dimgray", "Rectum" = "lightgray")
ccST <- apply(as.matrix(dimnames(dat)[[2]]), 1, vlookup, df=pdat, 
		col=which(names(pdat) == "Location"))
ccST <- coltypeST[as.character(ccST)]

# Analytical batch
table(pdat$Batch)
brewer.pal(9,"Set3")
coltypeB <- c("28"="#8DD3C7", "29" = "#FFFFB3", "30"="#BEBADA","33"="#FB8072",
			   "36"="#80B1D3", "41" = "#FDB462", "45"="#B3DE69","66"="#FCCDE5","46"="#D9D9D9")
ccB <- apply(as.matrix(dimnames(dat)[[2]]), 1, vlookup, df=pdat, 
		col=which(names(pdat) == "Batch"))
ccB <- coltypeB[as.character(ccB)]
ccB[is.na(ccB)] <- "#FFFFFF"

cc.col <- matrix(as.character(c(ccMC,ccST,ccB)), nrow=236,ncol=3)
colnames(cc.col) = c("Methylation Cluster", "Tumor Location","Analytical Batch")

temp.m <- dat

# Adjacent-normal samples
adjN.id <- pdat[pdat$TissueType=="Normal",1]
dat.adjN <- COADREAD_beta[ ,adjN.id]
dim(dat.adjN)
hmColors <- colorRampPalette(c("darkblue", "yellow"))(256)
@

<<label=fig1plot,include=FALSE>>=
pdf("HeatmpCustom.pdf")
heatmap.custom(
		temp.m, 		 
		x2 = dat.adjN, 
		order.x=NULL, 	 
		order.y=NULL, 
		Rowv = NULL, 
		Colv = NA,
		ColSideColors=cc.col,
		labCex=1.2, 
		scale = c("none"),
		labRow = "", 	
		labCol = "", 	
		na.rm = FALSE,
		col=hmColors,
		ratio=0.2, 		
		rowRatio=0.25, 	
		colRatio=0.08, 	
		margins = c(6,3),  
		sidemar=0.5,
		colLabSrt=75, 	
		rowColSrt=90, 	
		mtextCex=1.3,   
		mtext1="Colorectal Tumors", 
		mline1=3.1, 		
		mtext2="Normal", 
		mline2=3.1, 		
		mtextRow="DNA Methylation Probes (n=1403)", 
		keep.dendro=FALSE,
		Cdend=FALSE,
		Rdend=FALSE, 	
		verbose=FALSE)
dev.off()
@

\begin{figure}[ht!]
\begin{center}
\includegraphics{HeatmpCustom.pdf}
\caption{\label{fig:HeatmpCustom.pdf} RPMM-based classification and heatmap representation of DNA methylation data}
{\footnotesize{}}
\end{center}
\end{figure}

\newpage


\section*{Identification of candidate epigenetically silenced genes in colorectal cancer}
<<echo= false, results =hide>>=
setwd("/Users/Toshi/Desktop") # mac
#setwd("/home/toshi/Desktop") # linux
@

\subsection*{Array-based gene expression data}
The following level 3 gene expression data sets were downloaded from the TCGA Data Portal. COAD and READ expression data were then merged together. \\
\\
\noindent COAD expression data\\
unc.edu\_COAD.AgilentG4502A\_07\_3.Level\_3.1.5.0 \\
unc.edu\_COAD.AgilentG4502A\_07\_3.Level\_3.2.0.0 \\
unc.edu\_COAD.AgilentG4502A\_07\_3.mage-tab.1.7.0 \\

\noindent READ expression data\\
unc.edu\_READ.AgilentG4502A\_07\_3.Level\_3.1.1.0 \\ 
unc.edu\_READ.AgilentG4502A\_07\_3.Level\_3.2.0.0 \\
unc.edu\_READ.AgilentG4502A\_07\_3.mage-tab.1.2.0 \\

<<expression data>>=
load("unc.Agilent.COADREAD.merged.rda") 
@

<<echo= false, results =hide>>=
dat.exp <- unc.Agilent.COADREAD.ALL
@

<<expression data2>>=
dim(dat.exp)
dat.exp[1:5,1:4]
@

\subsection*{DNA methylation data}
<<methylation data>>=
dim(COADREAD_beta)
dim(COADREAD_pdata)
names(COADREAD_beta) <- substring(names(COADREAD_beta),1,15)
COADREAD_pdata$sampleID <- substring(COADREAD_pdata$sampleID,1,15)
dat.meth <- COADREAD_beta[,COADREAD_pdata$sampleID]
dim(dat.meth)
dat.meth[1:5,1:4]
@

\subsection*{Merge DNA methylation and expression data}
Identified the samples that are in common between DNA methylation and gene expression data.
<<intersect M and E>>=
length(intersect(names(dat.exp), names(dat.meth)))
intersect.ind <- which(COADREAD_pdata$sampleID %in% intersect(names(dat.exp), names(dat.meth)))
pdat.meth.exp <- COADREAD_pdata[intersect.ind,]
dim(pdat.meth.exp)
@

\noindent Various calculations were perfomed on DNA methylation data:\\

\noindent 1. Removed the NA-masked DNA methylation probes (n=4,484) and merged with the adf.
<<methylation calc>>=
count.na.row <- rowSums(is.na(dat.meth))
dat.meth$masked <- count.na.row==278
dat.meth <- dat.meth[dat.meth$masked=="FALSE",pdat.meth.exp$sampleID]
dim(dat.meth)

dat.meth.adf <- merge(adf.HM27, dat.meth, by="row.names")
row.names(dat.meth.adf) <- dat.meth.adf[[1]]
names(dat.meth.adf)[1] <- "ILmnID"  
dim(dat.meth.adf)
names(dat.meth.adf)[1:20]                                      
dat.meth.adf <- dat.meth.adf[,-c(7:11)]
dim(dat.meth.adf)

table(pdat.meth.exp$TissueType)
NormalID <- pdat.meth.exp$sampleID[pdat.meth.exp$TissueType=="Normal"]
TumorID <- pdat.meth.exp$sampleID[pdat.meth.exp$TissueType=="Tumor"]
@

\noindent 2. Calculated the mean $\beta$-values of tumor and adjacent-normal tissue samples.
<<methylation calc1>>=
dat.meth.adf$mean.betaN <- apply(dat.meth.adf[,NormalID],1,mean,na.rm=TRUE)
dat.meth.adf$mean.betaT <- apply(dat.meth.adf[,TumorID],1,mean,na.rm=TRUE)
@

\noindent 3. Calculated the $\beta$-values of 90th percentile in tumor sample.
<<methylation calc2>>=
dat.meth.adf$perc90.betaT <- apply(dat.meth.adf[,TumorID],1,quantile,probs=0.9,na.rm=TRUE)
@

\noindent 4. Calculated the $\beta$-value difference between 90th percentile in tumor vs. mean adjacent-normal tissue.
<<methylation calc3>>=
dat.meth.adf$perc90.betaT_diff <- dat.meth.adf$perc90.betaT-dat.meth.adf$mean.betaN 
@

\noindent Various calculations were also perfomed on gene expression data: 
<<expression calc1>>=
dat.exp <- dat.exp[,pdat.meth.exp$sampleID]
dim(dat.exp)
dat.exp <- cbind(row.names(dat.exp),dat.exp)
names(dat.exp)[1] <- "SYMBOL" 
@

\noindent 5. Calculated the mean expression values of tumor and adjacent-normal tissue samples.
<<expression calc2>>=
dat.exp$mean.expN <- apply(dat.exp[,NormalID],1,mean,na.rm=TRUE)
dat.exp$mean.expT <- apply(dat.exp[,TumorID],1,mean,na.rm=TRUE)
@

\noindent 6. Merged DNA methylation and gene expression data based on gene symbols.
<<merge>>=
identical(names(dat.meth.adf)[7:244],names(dat.exp)[2:239])
names(dat.meth.adf)[7:244] <- paste(names(dat.meth.adf[7:244]),"m",sep=".")
names(dat.exp)[2:239] <- paste(names(dat.exp[2:239]),"e",sep=".")
dat.meth.exp <- merge(dat.meth.adf, dat.exp, by="SYMBOL")
row.names(dat.meth.exp) <- dat.meth.exp[[2]]
dim(dat.meth.exp)
length(unique(dat.meth.exp$SYMBOL))
@

\noindent 7. Calculated the spearman correlation between DNA methylation and gene expression data.
<<merge corr>>=
meth <- names(dat.meth.exp)[7:244]
exp <- names(dat.meth.exp)[249:486]
identical(substring(meth,1,15),substring(exp,1,15))
spearman.corrlation <-function(x,s1,s2)
{
	x1 <- x[s1]
	x2 <- x[s2]
	x1 <- as.numeric(x1)
	x2 <- as.numeric(x2)
	cor(x1, x2, method = "spearman", use="complete.obs")
}
spearman.cor <- apply(dat.meth.exp, 1, spearman.corrlation, s1=exp, s2=meth)
summary(spearman.cor)
dat.meth.exp$spearman.cor <- spearman.cor
@

\subsection*{Identification of candidate epigenetically silenced genes}
<<epigenetically silenced genes>>=
sample.meth <- names(dat.meth.exp)[7:244]
sample.exp <- names(dat.meth.exp)[249:486]
dat.meth <- dat.meth.exp[,sample.meth]
dat.exp <- dat.meth.exp[,sample.exp]
identical(substring(names(dat.meth),1,15),substring(names(dat.exp),1,15))
table(pdat.meth.exp$TissueType)
TumorID <- pdat.meth.exp$sampleID[pdat.meth.exp$TissueType=="Tumor"]
@

DNA methylation data
<<epigenetically silenced Meth dat>>=
names(dat.meth) <- substring(names(dat.meth) ,1,15)
dat.methT <- dat.meth[,TumorID]
dim(dat.methT)
@

Gene expression data
<<epigenetically silenced Exp dat>>=
names(dat.exp) <- substring(names(dat.exp) ,1,15)
dat.expT <- dat.exp[,TumorID]
dim(dat.expT)
@

Calculated the mean gene expression corresponding to the samples with the highest 10$\%$ DNA methylation. 
<<90th percentile beta normal expression>>=
identical(names(dat.expT), names(dat.methT))
perc90.betaT <- dat.meth.exp$perc90.betaT
perc10M.expT <- vector("numeric",length=dim(dat.meth.exp)[[1]])
for (i in 1:dim(dat.meth.exp)[[1]]) {
	perc90.ind <- which(dat.methT[i,] > perc90.betaT[i])
	perc10M.expT[i] <- mean(as.numeric(dat.expT[i,perc90.ind]),na.rm=TRUE)
	#cat("Done", i, "out of",dim(dat.meth.exp)[[1]], "\n")
}
@

Log$_{2}$ gene expression change between normal and tumor with the highest 10$\%$ DNA methylation. 
<<expression diff>>=
dat.meth.exp$perc10M.expT <- perc10M.expT
dat.meth.exp$perc10.expT_diff <- dat.meth.exp$mean.expN-dat.meth.exp$perc10M.expT
dat.meth.exp.calc <- dat.meth.exp[,c(2,1,491,248,489,245,246)]
dim(dat.meth.exp.calc)
names(dat.meth.exp.calc)
attach(dat.meth.exp.calc,warn.conflicts = FALSE)
@

\noindent We considered the four criteria for an epigenetically silenced gene each with a relaxed and a stringent thresholds as described in the Supplemental Methods.\\

Relaxed criteria:
<<cutoffR>>=
filterR <- (perc10.expT_diff>log2(1.5)) + (perc90.betaT_diff > 0.1) + (spearman.cor<(-0.2)) + (mean.betaN<0.5)
@

Stringent criteria:
<<cutoffS>>=
filterS <- (perc10.expT_diff>log2(3)) + (perc90.betaT_diff > 0.3) + (spearman.cor<(-0.3)) + (mean.betaN<0.4)
@

Obtained a list of epigeneticallly silenced genes.
<<list>>=
dim(dat.meth.exp.calc[which(filterR==4 & filterS>=3),])
ES_Candidates <- dat.meth.exp.calc[which(filterR>=4 & filterS>=3),]
length(unique(ES_Candidates$SYMBOL))
as.character(sort(unique(ES_Candidates$SYMBOL)))
@

\newpage

\section*{\textit{MLH1} epigenetic silencing}
We assessed the \textit{MLH1} DNA methylation status in each sample based on the probe (cg00893636) located in the bidirectional \textit{MLH1/EPM2AIP1} promoter CpG island and closest to the current RefSeq \textit{MLH1} transcription start sites. This probe is annotated as \textit{EPM2AIP1} by Illumina. 
<<MLH1 and EPM2AIP1>>=
which(dat.meth.exp$SYMBOL=="MLH1")
which(dat.meth.exp$SYMBOL=="EPM2AIP1")
as.character(dat.meth.exp[c(6234),2])

pdat.meth.exp$TissueType <- factor(pdat.meth.exp$TissueType, level=c("Tumor","Normal"))
table(pdat.meth.exp$TissueType)
cor(as.numeric(dat.meth.exp[6234,meth]), 
		as.numeric(dat.meth.exp[11450,exp]), 
		method = "spearman", 
		use="complete.obs")

EPM2AIP1.cg00893636 <- cbind(data.frame(meth,as.numeric(dat.meth.exp[6234,meth]),
				exp,as.numeric(dat.meth.exp[6234,exp])), pdat.meth.exp)
names(EPM2AIP1.cg00893636)[c(2,4)] <- c("Meth.data","Exp.data")

p1 <- ggplot(EPM2AIP1.cg00893636, aes(Meth.data, Exp.data, color=TissueType))
p1.1 <- p1 + geom_point(size=2.2,alpha = 4/5)+ # shape=1
		scale_colour_manual(values = c("black","red"))+
		coord_cartesian(xlim = c(0,1))+
		scale_x_continuous("DNA Methylation") + scale_y_continuous("Gene Expression")+
		opts(title = "EPM2AIP1\nDNA methylation vs. Expression")

MLH1.cg00893636 <- cbind(data.frame(meth,as.numeric(dat.meth.exp[6234,meth]),
				exp,as.numeric(dat.meth.exp[11450,exp])), pdat.meth.exp)
names(MLH1.cg00893636)[c(2,4)] <- c("Meth.data","Exp.data")   

p2 <- ggplot(MLH1.cg00893636, aes(Meth.data, Exp.data, color=TissueType))
p2.1 <- p2 + geom_point(size=2.2,alpha = 4/5)+ # shape=1
		scale_colour_manual(values = c("black","red"))+
		coord_cartesian(xlim = c(0,1))+
		scale_x_continuous("DNA Methylation") + scale_y_continuous("Gene Expression")+
		opts(title = "MLH1\nDNA methylation vs. Expression")

<<label=fig2plot,include=FALSE>>=
		pdf("MLH1_Scatter.pdf")
pushViewport(viewport(layout = grid.layout(2, 1)))
vplayout <- function(x, y) viewport(layout.pos.row = x, layout.pos.col = y)
print(p1.1, vp = vplayout(1, 1))
print(p2.1, vp = vplayout(2, 1))
dev.off()
@

\begin{figure}[ht!]
\begin{center}
\includegraphics{MLH1_Scatter.pdf}
\caption{\label{fig:MLH1_Scatter.pdf} Scatter plot showing the relationship between DNA methylaiton and expression of \textit{MLH1} and \textit{EPM2AIP1}}
{\footnotesize{}}
\end{center}
\end{figure}

\subsection*{Identification of the \textit{MLH1} epigenetically silenced cases by k-means clustering method}
<<MLH1 kmeans>>=
names(MLH1.cg00893636)
Meth.data <- MLH1.cg00893636$Meth.data
head(Meth.data)
Exp.data <- MLH1.cg00893636$Exp.data
head(Exp.data)	
@

We re-scaled the \textit{MLH1} expression data between 0 and 1 (as is presented for DNA methylation $\beta$-values). We then performed K-means clustering (K=2) on the two-dimensional space of DNA methylation and expression data to classify the epigenetically silenced group and non-epigenetically silenced group of samples.
<<MLH1 kmeans2>>=
Meth.scale.a <- Meth.data/(max(Meth.data)-min(Meth.data))
Meth.scale.b <- Meth.scale.a-min(Meth.scale.a)
Exp.scale.a <- Exp.data/(max(Exp.data)-min(Exp.data))
Exp.scale.b <- Exp.scale.a-min(Exp.scale.a)

MLH1.scaled <- cbind(Meth.scale.b,Exp.scale.b)
row.names(MLH1.scaled) <- MLH1.cg00893636$sampleID
head(MLH1.scaled)

set.seed(1022)
cl <- kmeans(MLH1.scaled, 2)	
   
<<label=fig3plot,include=FALSE>>=
pdf("MLH1_kmeans.pdf")
plot(MLH1.scaled, col = cl$cluster,ylab='Expression', xlab='DNA methylation')
points(cl$centers, col = 1:2, pch = 8, cex=2)
dev.off()
@

\begin{figure}[ht!]
\begin{center}
\includegraphics{MLH1_kmeans.pdf}
\caption{\label{fig:MLH1_kmeans.pdf} Identification of \textit{MLH1} epigenetically silenced cases. Black open circles indicate cases with \textit{MLH1} epigenetic silencing. Red open circles indicate cases without \textit{MLH1} epigenetic silencing. Cluster centriods are indicated by the stars.}
{\footnotesize{}}
\end{center}
\end{figure}

\newpage

<<sessionInfo>>=
sessionInfo()
@

\end{document}